src/openrct2/paint/Supports.h

```diff
@@ -93,6 +93,9 @@ enum class MetalSupportType : uint8_t
      */
     BoxedCoated,
 };
+constexpr uint8_t MetalSupportTypeCount = 13;
+
+extern MetalSupportType RotatedMetalSupports[MetalSupportTypeCount][NumOrthogonalDirections];
```

Comment:
Not a fan of exposing data in headers via extern, I think its best to some of the paint functions that do the supports into Supports.cpp, most of the TrackPaintUtil functions are mostly wrappers that do nothing special but draw the supports based on the direction. The actual drawing code for the supports should become static functions and the Util ones should be the ones exposed in the header inside Supports.h.

Reply:
Right, I see what you mean. But that would mean putting general support functions (the stuff already in `Supports.cpp`) together with track drawing code (`TrackPaintUtilDrawStationMetalSupports()`). To me, that seems less desirable than exposing `RotatedMetalSupports`. (It also seemed more in line with how the rest of the code is structured.)

Is there perhaps another way that is a bit cleaner?

Reply:
What I mean is that all code that is painting supports should end up in Supports.cpp, we don't have to leak such data for supports into the track painting code, any TrackPaintUtil* function that is just drawing supports should be also moved to Supports.cpp and possibly renamed.

Reply:
Ok, I made the changes you requested.